EpikCord/client/http.py

```diff
@@ -0,0 +1,475 @@
+                "User-Agent": f"DiscordBot (https://github.com/EpikCord/EpikCord.py {__version__})",
+            },
+            json_serialize=lambda x, *__, **___: json.dumps(x).decode("utf-8")  # type: ignore
+            if _ORJSON
+            else json.dumps(x),
+            ws_response_class=GatewayWebsocket,
+        )
+        self.buckets: Dict[str, Union[Bucket, TopLevelBucket]] = {}
+        self.global_event: asyncio.Event = asyncio.Event()
+        self.global_event.set()
+
+    async def ws_connect(
+        self, url: str, headers: Dict = {}, **kwargs
+    ) -> aiohttp.ClientWebSocketResponse:
+        headers.update(
+            {
+                "User-Agent": f"DiscordBot (https://github.com/EpikCord/EpikCord.py {__version__})"
+            }
+        )
+        return await self.session.ws_connect(url, headers=headers, **kwargs)
+
+    async def request(
+        self,
+        route: Route,
+        *args,
+        discord: bool = True,
+        json: Optional[Dict] = None,
+        files: Optional[List[File]] = None,
+        **kwargs,
+    ) -> aiohttp.ClientResponse:
+        """
+        Parameters
+        ----------
+        *args
+            The args to pass to the ClientSession.request method.
+        discord: bool
+            Whether this request is to Discord.
+        json: Optional[Dict]
+            The json to pass to the ClientSession.request method.
+        files: Optional[List[File]]
+            The files to pass to the request.
+        **kwargs
+            The kwargs to pass to the ClientSession.request method.
+
+        Returns
+        -------
+        aiohttp.ClientResponse
+            The response of the request.
+
+        Raises
+        ------
+        NotFound
+            If the request returns a 404 status code.
+        Forbidden
+            If the request returns a 403 status code.
+        Unauthorized
+            If the request returns a 401 status code.
+        BadRequest
+            If the request returns a 400 status code.
+        HTTPException
+            If the request returns a status code that is not OK and not any of the other exceptions.
+        """
+
+        method = route.method
+        url = route.url
+
+        if not discord:
+            return await self.session.request(method, url, *args, **kwargs)
+
+        self.setup_kwargs(kwargs, files=files, json=json)
+
+        url = self.clean_url(url)
+        bucket = self.buckets.get(f"{method}:{url}") or MockBucket()
+
+        await self.global_event.wait()
+        await bucket.wait()
+
+        for _ in range(5):
+
+            async with self.session.request(method, url, *args, **kwargs) as response:
+                if isinstance(bucket, MockBucket):
+                    bucket = await self.set_bucket(
+                        response=response,
+                        **route.major_parameters,
+                    )
+
+                data = await self.extract_content(response)
```

Comment:
This should be done in `handle_ratelimit` instead.

Reply:
But that isn't the only place I need it, I need it in `log_request`, unless you have some other point